browser_patches/firefox/juggler/TargetRegistry.js

```diff
@@ -657,6 +650,29 @@ function dirPath(path) {
   return path.substring(0, path.lastIndexOf('/') + 1);
 }
 
+async function waitForWindowReady(window) {
+  if (window.delayedStartupPromise) {
+    await window.delayedStartupPromise;
+  } else {
+    await new Promise((resolve => {
+      Services.obs.addObserver(function observer(aSubject, aTopic) {
+        if (window == aSubject) {
+          Services.obs.removeObserver(observer, aTopic);
+          resolve();
+        }
+      }, "browser-delayed-startup-finished");
```

Comment:
IIUC, this could only work for the first default window out there. Should we only run this code for the default context?

Reply:
This happen for both default and non-default contexts. Interestingly, this is the reason why we were getting that crazy number-error during our today's zoom debugging session - I was later getting it for some other tests, so had to debug.

Overall, it looks like this happens for the top-level windows, e.g. `window.open` would give a window that fires the same event. At least I saw some tests rely on it: https://ffsearch.azurewebsites.net/#path=%2Fhome%2Fjoe%2Ffirefox%2Fbrowser%2Fbase%2Fcontent%2Ftest%2Fperformance%2Fbrowser_windowopen.js&line=121